src/app/shell/personal-cabinet/shared-cabinet/messages/chat/chat.component.ts

```diff
@@ -81,60 +81,100 @@ export class ChatComponent implements OnInit, AfterViewInit, OnDestroy {
 
   public ngAfterViewInit(): void {
     this.setChatSubscriptions();
+  }
 
-    this.chatEl.nativeElement.onscroll = () => {
-      if (this.chatEl.nativeElement.scrollTop < 10) {
-        this.chatEl.nativeElement.scrollTop = 10;
-
-        if (!this.isHistoryLoading) {
-          this.isHistoryLoading = true;
-          this.messagesParameters.from = this.messages.length;
-          if (this.mode === ModeConstants.WORKSHOP) {
-            this.store.dispatch(new GetChatRoomMessagesByWorkshopId(this.chatRoom.workshopId, this.messagesParameters));
-          } else {
-            this.store.dispatch(new GetChatRoomMessages(this.chatRoom.id, this.userRole, this.messagesParameters));
+  public ngOnDestroy(): void {
+    this.store.dispatch([new PopNavPath(), new ClearSelectedChatRoom(), new ResetProviderWorkshopDetails()]);
+    this.hubConnection.stop();
+    this.destroy$.next(true);
+    this.destroy$.unsubscribe();
+  }
+
+  private setChatSubscriptions(): void {
+    this.chatRoom$.pipe(filter(Boolean), takeUntil(this.destroy$)).subscribe((chatRoom: ChatRoom) => {
+      this.chatRoom = chatRoom;
+      this.store.dispatch(new GetChatRoomMessagesById(this.userRole, this.chatRoom.id, this.messagesParameters));
+      this.getChatMembersNames();
+
+      this.chatEl.nativeElement.onscroll = () => {
```

Comment:
use `addEventListener`, so we can call `removeEventListener` on destroy
another approach is Angular `Renderer2 tool